sound/soc/sof/intel/hda.c

```diff
@@ -1176,8 +1176,8 @@ int hda_dsp_remove(struct snd_sof_dev *sdev)
 				SOF_HDA_INT_CTRL_EN | SOF_HDA_INT_GLOBAL_EN, 0);
 
 	/* disable cores */
-	if (chip)
-		hda_dsp_core_reset_power_down(sdev, chip->host_managed_cores_mask);
+	if (chip && chip->power_down_dsp)
+		chip->power_down_dsp(sdev);
```

Comment:
if you are not testing the result, should the op return void?

Reply:
I reviewed the code. I will keep the return type and check the return value. But it won't return from here even with the error, it need to fall through to the end. I will put that comment also. Will update the patch soon.